matrix_common/servlet.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import json
+import logging
+from typing import Any, Callable
+
+from twisted.internet import defer
+from twisted.python import failure
+from twisted.web import server
+from twisted.web.server import Request
+
+from matrix_common.json import JsonDict
+
+logger = logging.getLogger(__name__)
+
+
+class MatrixRestError(Exception):
+    """
+    Handled by the jsonwrap wrapper. Any servlets that don't use this
```

Comment:
Does `jsonwrap` refer to the decorators below?

Reply:
Err, yes, I forgot to rename it here.